### PR TITLE
improving ulimit docs

### PR DESCRIPTION
**Pull request by asoorm**

### CODE REVIEW

File: tyk-docs/content/deploy-tyk-premise-production/deploy-tyk-premise-production.mmark [hunk 1]
```diff
@@ -188,11 +188,37 @@ Tyk is CPU-bound, you will get exponentially better performance the more cores y
...
 
-Make sure your system has resource limits set to handle lots of inbound traffic.
+Make sure your system has resource limits set to handle an appropriate number of connections.
+We shall assume your operating system is using `systemd`, the default init system for all major Linux distributions.
 
-{{./static/include/file-handles.md}}
+Simply modify your `systemd` unit files for each of the Tyk services.
 
-Careful with it though, it could lead to unterminated connections.
+* Gateway `/usr/lib/systemd/system/tyk-gateway.service`
```
Comment (excieve) [inline tyk-docs/content/deploy-tyk-premise-production/deploy-tyk-premise-production.mmark]: Shouldn't recommend modifying these as they'll get overwritten on package upgrade. Instead suggest systemd unit overrides (`systemctl edit`).

**Review by excieve — CHANGES_REQUESTED** [review]
In addition to this, possibly mention how to change it for docker.

**Comment by asoorm** [comment]
@excieve PTAL

File: tyk-docs/content/deploy-tyk-premise-production/deploy-tyk-premise-production.mmark [hunk 2]
```diff
@@ -188,11 +188,45 @@ Tyk is CPU-bound, you will get exponentially better performance the more cores y
...
-Careful with it though, it could lead to unterminated connections.
+```
+fs.file-max=160000
+```
+
+Please note that a single file, with associated inode & dcache consumes approximately 1kb RAM. As such, setting
+`fs.file-max=160000` will consume a maximum of 160Mb ram.
+
+The changes will apply after a system reboot, but if you do not wish to reboot quite yet, you can apply the change
+for the current session using `echo 120000 > /proc/sys/fs/file-max`.
```
Comment (excieve) [inline tyk-docs/content/deploy-tyk-premise-production/deploy-tyk-premise-production.mmark]: `160000`?

File: tyk-docs/content/deploy-tyk-premise-production/deploy-tyk-premise-production.mmark [hunk 3]
```diff
@@ -188,11 +188,45 @@ Tyk is CPU-bound, you will get exponentially better performance the more cores y
...
 
-{{./static/include/file-handles.md}}
+You can increase the maximum number of files available to the kernel by modifying `/etc/sysctl.conf`.
 
-Careful with it though, it could lead to unterminated connections.
+```
+fs.file-max=160000
+```
+
+Please note that a single file, with associated inode & dcache consumes approximately 1kb RAM. As such, setting
```
Comment (excieve) [inline tyk-docs/content/deploy-tyk-premise-production/deploy-tyk-premise-production.mmark]: Nit: `KB` or `kB` and `MB` (`kb` is kilobit and `Mb` is megabit) :)

**Review by excieve — COMMENTED** [review]
I'd still add a note for containers, even if it's just about changing the container runtime's (such as Docker) unit file.

**Review by excieve — APPROVED** [review]
LGTM

**Pull request merged**